### PR TITLE
fix(lexical): missing style on the list when user copy and paste from other sources

### PR DESCRIPTION
**Pull request by mihajlovco**
With this PR we fixed the problem of missing style on the list when user copy and paste from other sources.

## Changes
- Changes `WebinyListNode` to intercept and add theme styles to the list by default.
- Fallback to default style in case the user has removed the previously used style.
> For example if I have now style with an id list, and remove that one, the node on creation and update will try to find the first occurrence of the style list for the specified `ol` or `ul` tag. 

## How Has This Been Tested?
Manually.

### CODE REVIEW

File: packages/lexical-editor/src/nodes/list-node/BaseListNode.ts
Comment (adrians5j) [inline packages/lexical-editor/src/nodes/list-node/BaseListNode.ts:1]: Let's move this file up, into `nodes` folder, and have it import additional code from the `list-node` folder. But let's rename the `list-node` folder to `ListNode`, and, in the same way we did it in prev PRs, let's rename `BaseListNode` to `ListNode`.

**Review by adrians5j — CHANGES_REQUESTED** [review]
For starters, let's make things a bit more consistent.

File: packages/lexical-editor/src/utils/nodes/listNode.ts
Comment (adrians5j) [inline packages/lexical-editor/src/utils/nodes/listNode.ts:1]: If this is only related to `ListNode`, I don't see the reason why this should be located outside of the `nodes` folder. Is this only present in `ListNode` ?

File: packages/lexical-editor/src/components/ToolbarActions/QuoteAction.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ export const QuoteAction = () => {
             formatToQuote(editor, hasQuoteStyles ? DEFAULT_QUOTE_ID : undefined);
             return;
         }
+        debugger;
```
Comment (adrians5j) [inline packages/lexical-editor/src/components/ToolbarActions/QuoteAction.tsx]:
```suggestion

```

**Review by adrians5j — CHANGES_REQUESTED** [review]
Did a quick review, still need to actually test it in browser.

File: packages/lexical-editor/src/components/ToolbarActions/QuoteAction.tsx (hunk 1 above)
Comment (mihajlovco) [reply packages/lexical-editor/src/components/ToolbarActions/QuoteAction.tsx]: removed thanks

File: packages/lexical-editor/src/nodes/ListNode.ts [hunk 2]
```diff
@@ -40,27 +41,40 @@ export class WebinyListNode extends ElementNode {
...
+        this.__start = start || 1;
     }
 
     static override getType() {
         return "webiny-list";
     }
 
+    /*
+     *
+     * */
```
Comment (adrians5j) [inline packages/lexical-editor/src/nodes/ListNode.ts]:
```suggestion

```

File: packages/lexical-editor/src/nodes/ListNode.ts [hunk 3]
```diff
@@ -40,27 +41,40 @@ export class WebinyListNode extends ElementNode {
...
     override createDOM(config: EditorConfig): HTMLElement {
         const tag = this.__tag;
         const dom = document.createElement(tag);
+        const wTheme = config.theme as WebinyTheme;
 
         if (this.__start !== 1) {
             dom.setAttribute("start", String(this.__start));
         }
 
+        // if styleId is not set or user removed from theme, set default style.
```
Comment (adrians5j) [inline packages/lexical-editor/src/nodes/ListNode.ts]:
```suggestion
        // If styleId is not set or user removed from theme, set default style.
```

File: packages/lexical-editor/src/nodes/ListNode.ts [hunk 4]
```diff
@@ -137,25 +146,56 @@ export class WebinyListNode extends ElementNode {
...
+        if (!tag) {
+            return;
+        }
+
+        const themeEmotionMap = theme?.emotionMap;
+        if (!themeEmotionMap) {
+            return;
+        }
+
+        const style = findTypographyStyleByHtmlTag(tag, themeEmotionMap);
```
Comment (adrians5j) [inline packages/lexical-editor/src/nodes/ListNode.ts:159]: I noticed this doesn't work for ordered list. Reported it here https://webiny.slack.com/archives/C01BL618QLA/p1685357931418519

**Review by adrians5j — CHANGES_REQUESTED** [review]

File: packages/lexical-editor/src/nodes/ListNode.ts (hunk 4 above)
Comment (mihajlovco) [reply packages/lexical-editor/src/nodes/ListNode.ts:159]: Functionally works in the correct way, if there is no theme style the `typography action will be removed` from the toolbar. 
We currently don't have a theme style for the ordered list. That would require migration and we will postpone for another version.

File: packages/lexical-editor/src/nodes/ListNode.ts (hunk 2 above)
Comment (mihajlovco) [reply packages/lexical-editor/src/nodes/ListNode.ts]: done

File: packages/lexical-editor/src/nodes/ListNode.ts (hunk 3 above)
Comment (mihajlovco) [reply packages/lexical-editor/src/nodes/ListNode.ts]: done

**Review by adrians5j — APPROVED** [review]

**Pull request merged**